lib/src/model/broadcast/broadcast_controller.dart

```diff
@@ -0,0 +1,126 @@
+import 'dart:async';
+
+import 'package:dartchess/dartchess.dart';
+import 'package:deep_pick/deep_pick.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:lichess_mobile/src/model/broadcast/broadcast.dart';
+import 'package:lichess_mobile/src/model/broadcast/broadcast_repository.dart';
+import 'package:lichess_mobile/src/model/common/chess.dart';
+import 'package:lichess_mobile/src/model/common/http.dart';
+import 'package:lichess_mobile/src/model/common/id.dart';
+import 'package:lichess_mobile/src/model/common/socket.dart';
+import 'package:lichess_mobile/src/utils/json.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'broadcast_controller.g.dart';
+
+@riverpod
+class BroadcastController extends _$BroadcastController {
+  static Uri broadcastSocketUri(BroadcastRoundId broadcastRoundId) =>
+      Uri(path: 'study/$broadcastRoundId/socket/v6');
+
+  StreamSubscription<SocketEvent>? _subscription;
+
+  late SocketClient _socketClient;
+
+  @override
+  Future<BroadcastMapGames> build(BroadcastRoundId broadcastRoundId) async {
+    _socketClient = ref
+        .read(socketPoolProvider)
+        .open(BroadcastController.broadcastSocketUri(broadcastRoundId));
+
+    _subscription = _socketClient.stream.listen(_handleSocketEvent);
+
+    ref.onDispose(() {
+      _subscription?.cancel();
+    });
+
+    return await ref.withClient(
+      (client) => BroadcastRepository(client).getRound(broadcastRoundId),
+    );
+  }
+
+  void _handleSocketEvent(SocketEvent event) {
+    if (!state.hasValue) return;
+
+    switch (event.topic) {
+      // Sent when a node is recevied from the broadcast
+      case 'addNode':
+        _handleAddNodeEvent(event);
+      // Sent when a game ends
+      case 'chapters':
+        _handleChaptersEvent(event);
+      // Sent when clocks are updated from the broadcast
+      case 'clock':
+        _handleClockEvent(event);
+    }
+  }
+
+  void _handleAddNodeEvent(SocketEvent event) {
+    final relayPath = pick(event.data, 'p', 'relayPath').asUciPathOrThrow();
```

Comment:
Explain with comments what is the difference between `relayPath` and `path`. Since I cannot figure it out while reviewing, I guess it will be the same for anyone reading this code.